nixos/modules/services/network-filesystems/kubo.nix

```diff
@@ -5,6 +5,27 @@ let
 
   settingsFormat = pkgs.formats.json {};
 
+  rawDefaultConfigFile = pkgs.runCommand "kubo-default-config" {
+    nativeBuildInputs = [ cfg.package ];
+  } ''
+    export IPFS_PATH="$TMPDIR"
+    ipfs init --empty-repo --profile=${profile}
+    ipfs --offline config show > "$out"
+  '';
+
+  rawDefaultConfig = (builtins.fromJSON (builtins.readFile rawDefaultConfigFile));
+
+  # Delete the "Identity" section since it only contains the PeerID of the
+  # temprary IPFS repo above and is thus unrelated to the actual config of the daemon.
+  # It is also randomly generated each time and is thus a source of non-determinism.
+  # The "Pinning" section contains the "RemoteServices" section, which would prevent
+  # the daemon from starting as that setting can't be changed via ipfs config replace.
+  defaultConfig = builtins.removeAttrs rawDefaultConfig [ "Identity" "Pinning" ];
+
+  customizedConfig = lib.recursiveUpdate defaultConfig cfg.settings;
+
+  configFile = settingsFormat.generate "kubo-config.json" customizedConfig;
```

Comment:
I wouldn't mind if we would reduce the variables to one or two less.

Reply:
By using `lib.importJSON`, the line became short enough that I could combine it with another line without it becoming too long.